### DISCUSSION TITLE
Deploying on lighttpd

### DISCUSSION THREAD

**Discussion by master0v** [miguelgrinberg/Flask-SocketIO#1689] [Q&A]
I am currently using flup to deploy my Flask app on lighttpd.  The code is as follows:

```
#!/usr/bin/python3
from flup.server.fcgi import WSGIServer
from switches import app

if __name__ == '__main__':
    WSGIServer(app).run() #app
```

Once I introduced flask_socketio, `socketio = SocketIO(app)` I can no longer use the same model.

I tried the naive approach as follows

```
#!/usr/bin/python3
from flup.server.fcgi import WSGIServer
from switches import socketio

if __name__ == '__main__':
    WSGIServer(socketio).run() #app
```

but of course that didn't work.  How do I change my fcgi wrapper to make it work with flask_socketio?
any advice is really appreciated!

**Comment by miguelgrinberg** [comment]
Lighttpd is not a supported web server. I really don't know if it can be configured to work with Socket.IO. For starters, I don't think FastCGI or flup support WebSocket, so in the best scenario you'll be limited to use long-polling only.

**Reply by master0v** [reply]
Thanks for your answer! I got it working, like you said with long-polling.  It is ok for now, but hopefully the proper support will come in the future.

**Reply by miguelgrinberg** [reply]
I doubt there'll be any new development around FastCGI, as it is an outdated solution. I recommend you migrate to one of the supported deployment methods described in the documentation.

**Reply by master0v** [reply]
Hey Miguel, thanks for your reply.  Just to clarify, when you say "supported deployment methods" you mean - a web server that is supported, right?  The problem is not with lighttpd, because lighttpd 1.4.46 and later support websockets as follows:

Docs_ModWStunnel provides a websocket tunnel through which client and a backend server can tunnel another protocol.
Docs_ModProxy proxies to a backend websocket server
Docs_ModCGI can run a backend CGI to act as websocket server for a request

I just don't know those things well enough to be able to configure them.

**Reply by miguelgrinberg** [reply]
The WSGI protocol does not natively support WebSocket, there are a few different extensions to this protocol that people have created. The Flask-SocketIO server supports a few of them, all listed in the documentation. The fact that Lighttpd supports WebSocket does not really help, because the WebSocket needs to be given to the Python application, and FastCGI does not have a mechanism to do that.

If you are really attached to Lighttpd, maybe you'll be able to use it as a reverse proxy, with one of the supported web servers. Basically you would replace nginx with Lighttpd. But you'll have to drop FastCGI for sure.

**Reply by master0v** [reply]
Sorry for the dumb question: is FastCGI  and ModCGI the same thing?

this line "Docs_ModCGI can run a backend CGI to act as websocket server for a request" creates an impression that somehow I can have my socketio app be a child process which handles the socket.io requests.

**Reply by miguelgrinberg** [reply]
I don't really know. CGI is not the same as FastCGI, so my guess is that they are different.

In any case, as I said before, there is no standard mechanism by which a web server passes a WebSocket to a Python application, so just because Lighttpd can do it does not mean it'll work, because there is nothing on the Python side that is compatible with Lighttpd. The only option that you have is to use Lighttpd as a reverse proxy, I mentioned that above. So no FastCGI, no ModCGI, no flup, none of that. That is all incompatible.
